src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -319,6 +345,38 @@ module Make (Inputs : Inputs_intf) :
     let node = Hashtbl.find_exn t.table hash in
     node.breadcrumb
 
+  let path_search t state_hash ~find ~f =
```

Comment:
We don’t need a path longer than k, right? (Maybe when we are testing with a very small k, but in production, it shouldn’t be necessary?) if I’m right about this, should we cap path length at all?

Reply:
The length is capped in the next PR. But it's capped to 2k. I did this because there were instances in the test network were nodes where joining and they needed to catch up. But the path of transitions needed to be 1.5K.